### PR TITLE
feat: use access api to hide inaccessible content

### PR DESCRIPTION
**Pull request by anho**

### CODE REVIEW

File: packages/reference/src/common/MultipleReferenceEditor.tsx [hunk 1]
```diff
@@ -92,6 +85,11 @@ function Editor(props: EditorProps) {
     [linkActionsProps]
   );
 
+  const testProps = {
```
Comment (z0al) [inline packages/reference/src/common/MultipleReferenceEditor.tsx:88]: Is the naming here intentional ? :)

File: packages/reference/src/common/useContentTypePermissions.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    async function checkContentTypeAccess() {
+      const creatable = await getContentTypes('create');
+      const readable = await getContentTypes('read');
+      setCreatableContentTypes(creatable);
+      setReadableContentTypes(readable);
+    }
+
+    void checkContentTypeAccess();
+  }, [availableContentTypes]);
```
Comment (z0al) [inline packages/reference/src/common/useContentTypePermissions.ts:64]: I think the `useEffect` dependency list needs to be updated, oder?

**Review by z0al — COMMENTED** [review]
Took a quick look. Looking good. Will review it again after the power hour

File: packages/reference/src/common/MultipleReferenceEditor.tsx (hunk 1 above)
Comment (anho) [reply packages/reference/src/common/MultipleReferenceEditor.tsx:88]: 🦅 👀

**Review by suevalov — COMMENTED** [review]
Looks good, did you test it with a local copy of the main web app?

File: packages/reference/src/common/useContentTypePermissions.ts (hunk 2 above)
Comment (anho) [reply packages/reference/src/common/useContentTypePermissions.ts:64]: It is fine like this.

**Comment by anho** [comment]
> Looks good, did you test it with a local copy of the main web app?

Quick sanity check yes. Will integrate it back after an actual testing session.

**Review by suevalov — APPROVED** [review]

File: packages/reference/src/common/useContentTypePermissions.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    if (props.validations.contentTypes) {
+      return props.allContentTypes.filter((ct) =>
+        props.validations.contentTypes?.includes(ct.sys.id)
+      );
+    }
+
+    return props.allContentTypes;
+  }, [props.validations.contentTypes, props.entityType]);
```
Comment (djagya) [inline packages/reference/src/common/useContentTypePermissions.ts:46]: Did you intentionally leave `allContentTypes` out of deps array?

**Review by giotiskl — APPROVED** [review]

**Review by djagya — APPROVED** [review]
Nice. The `.mdx` files used by docz don't need to be updated, right? (like `SingleEntryReferenceEditor.mdx`)

File: packages/reference/src/common/useContentTypePermissions.ts (hunk 3 above)
Comment (anho) [reply packages/reference/src/common/useContentTypePermissions.ts]: Nope, I see a pattern where I miss out on updating effect dependencies 😄

**Comment by anho** [comment]
@djagya 
> Nice. The .mdx files used by docz don't need to be updated, right? (like SingleEntryReferenceEditor.mdx)

No need to update, there are no external changes.

**Pull request merged**